pyccel/codegen/printing/ccode.py

```diff
@@ -586,6 +602,68 @@ def function_signature(self, expr):
         else:
             return '{0}{1}({2})'.format(ret_type, name, arg_code)
 
+    def _print_IndexedElement(self, expr):
+        if isinstance(expr.base, IndexedVariable):
+            base = expr.base.internal_variable
+        else:
+            base = expr.base
+        base = self._print(expr.base.label)
+        inds = list(expr.indices)
+        inds = inds[::-1]
+        base_shape = Shape(expr.base)
+        allow_negative_indexes = (isinstance(expr.base, IndexedVariable) and \
+                expr.base.internal_variable.allows_negative_indexes)
+        for i, ind in enumerate(inds):
+            if isinstance(ind, PyccelUnarySub) and isinstance(ind.args[0], Integer):
+                inds[i] = PyccelMinus(base_shape[i], ind.args[0])
+            else:
+                if isinstance(ind, Slice):
+                    #setting the slice start and end to their correct value if none is provided
+                    start = ind.start
+                    end = ind.end
+                    if ind.start is None:
+                        start = 0
+                    if ind.end is None:
+                        end = expr.base.shape[i]
+                    inds[i] = Slice.__new__(Slice, start, end)
```

Comment:
You don't need to call the `__new__` method explicitly. And you probably shouldn't. I don't know if the `__init__` method is called if you do this. To declare an instance of a class you should just do:
```
inds[i] = Slice(start, end)
```